src/Common/tests/System/PlatformDetection.cs

```diff
@@ -23,6 +23,32 @@ public static class PlatformDetection
 
         public static int WindowsVersion { get; } = GetWindowsVersion();
 
+        public static bool IsWindowsSubsystemForLinux
```

Comment:
Given how many tests will be checking this now and that it's doing file I/O, I wonder if we should cache it?